airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/availability_strategy.py

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from deprecated.classic import deprecated
+
+
+class StreamAvailability(ABC):
+    @abstractmethod
+    def is_available(self) -> bool:
```

Comment:
Given how we're just using these two methods to assemble a tuple that we return in the `check_availability()` > `return stream_availability.is_available(), stream_availability.message()`, can `is_available()` and `message()` just be consolidated into one method `is_available()` that returns a tuple of both bool + Optional[str]?

Reply:
+1 to what Maxime said. I intentionally created a class to replace the Tuple because tuples are difficult to evolve